paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -547,32 +557,92 @@ def __init__(
         self.block = block
         self.soa_dir = soa_dir
         self.timeout = timeout
-
         self.mark_for_deployment_return_code = -1
         self.wait_for_deployment_green_light = Event()
+        self.retval = 0
+
+        self.human_readable_status = "Waiting on mark-for-deployment to initialize..."
+        self.progress = Progress()
+        self.slack_client = get_slack_client()
+        self.slack_ts = None
+        self.slack_channel = "#test"
+        self.slack_channel_id = None
+        self.active_button = None
+        self.available_buttons = []
```

Comment:
non-blocking thought:

maybe instead of having various `on_enter_*` methods update a list, we should look up available_buttons from a dictionary keyed on state?